Services/UI/OJS.Services.Ui.Business/Implementations/ParticipantsBusinessService.cs

```diff
@@ -168,7 +168,8 @@ private static void AssignRandomProblemsToParticipant(Participant participant, C
         var random = new Random();
 
         var problemGroups = contest.ProblemGroups
-            .Where(pg => !pg.IsDeleted && pg.Problems.Any(p => !p.IsDeleted));
+            .Where(pg => !pg.IsDeleted && pg.Problems.Any(p => !p.IsDeleted))
+            .OrderBy(pg => pg.OrderBy);
```

Comment:
Why order the groups here? I don't think we need them ordered to choose one random.